source/common/http/filter_manager.cc

```diff
@@ -1065,8 +1067,8 @@ void FilterManager::encodeHeaders(ActiveStreamEncoderFilter* filter, ResponseHea
   }
 
   const bool modified_end_stream = (end_stream && continue_data_entry == encoder_filters_.end());
-  state_.non_100_response_headers_encoded_ = true;
   filter_manager_callbacks_.encodeHeaders(headers, modified_end_stream);
+  state_.non_100_response_headers_encoded_ = true;
```

Comment:
this change is necessary since there's a check of `non_100_response_headers_encoded_ != false` in the local reply.